src/meta/src/barrier/rpc.rs

```diff
@@ -109,17 +110,33 @@ impl ControlStreamManager {
             .committed_epoch;
         let node_id = node.id;
         let node_host = node.host.clone().unwrap();
-        match self.context.new_control_stream_node(node, prev_epoch).await {
-            Ok((stream_node, response_stream)) => {
-                let _ = self.nodes.insert(node_id, stream_node);
-                self.response_streams
-                    .push(into_future(node_id, response_stream));
-                info!(?node_host, "add control stream worker");
-            }
-            Err(e) => {
-                error!(err = %e.as_report(), ?node_host, "fail to start control stream with worker node");
+        let mut backoff = ExponentialBackoff::from_millis(100)
+            .max_delay(Duration::from_secs(3))
+            .factor(5);
+        const MAX_RETRY: usize = 5;
+        for i in 1..=MAX_RETRY {
+            match self
+                .context
+                .new_control_stream_node(node.clone(), prev_epoch)
+                .await
+            {
+                Ok((stream_node, response_stream)) => {
+                    let _ = self.nodes.insert(node_id, stream_node);
+                    self.response_streams
+                        .push(into_future(node_id, response_stream));
+                    info!(?node_host, "add control stream worker");
+                    return;
+                }
+                Err(e) => {
+                    // It may happen that the dns information of newly registered worker node
+                    // has not been propagated to the meta node and cause error. Wait for a while and retry
```

Comment:
Prior to the bidi stream refactoring, we didn't see this issue in our tests. Is it because previously the creation of the RPC connection is delayed until the first barrier injection given that it is a unary RPC while after the refactoring the creation of the RPC connection comes earlier when the worker node is added?

Reply:
Yes, I think so. With this PR the meta node connects to CN earlier than before.